### ISSUE TITLE
Perpetually repeating notifications

### ISSUE THREAD

**Issue by safris** [sargsyan/github-notifier#19]
**Describe the bug**
I have configured github.com as the source of GitHub notification events. Though I successfully receive notifications, the same notification is re-posted multiple times. The only way to stop the notifications from recurring is to go to the Notification Center in GitHub and mark all notifications as read.

**To Reproduce**
1. Receive a notification.
1. Continue to receive the same notification.
1. Go to Notification Center in GitHub and mark all notifications as read to stop receiving the same notification.

**Expected behavior**
I expect that the notification for any particular event is posted only once.

**Screenshots**
<img width="337" alt="Screen Shot 2020-02-11 at 10 31 34 AM" src="https://user-images.githubusercontent.com/1258414/74266776-b7879c00-4cb9-11ea-9e1b-b35e94aaffe5.png">

**Desktop (please complete the following information):**
 - OSX version: MacOS 10.15.3

**Additional context**
None

**Comment by sargsyan** [comment]
Hi @safris 

This can happen when ~/.latest_shown file's permission is changed and it's not writeable for your user. I experienced this sometimes after I updated the macOS version. I don't know the root cause yet, but manually fixing the changed permission (adding +w permission for the owner) fixed the issue for me.

**Comment by safris** [comment]
Hi @sargsyan, thank you for your response. I checked the permissions of the file, and they seem to be correct:
```
$ ll ~/.latest_shown
-rw-r--r--  1 seva  staff  30 Feb 13 07:31 /Users/seva/.latest_shown
```

**Comment by sargsyan** [comment]
Hi @safris,

I updated my MacOS to version 10.15.3 today, but I wasn't able to see the issue. Could you look at the error messages in the application log file and share the latest ERROR messages if there any? You can do it with 

`tail ~/Library/Logs/github_notif/service.log`

**Comment by safris** [comment]
Hi @sargsyan, here is the output from `service.log`:
```
2020-02-14 19:10:52 ERROR Cannot save last shown commit date. 2020-02-14 19:10:52 ERROR An error occured while showing notification. An error occured while accessing https://api.github.com/repos/lightstep/community-team/issues/45: Not Found
1581719798
2020-02-14 19:11:53 ERROR Cannot save last shown commit date. 2020-02-14 19:11:53 ERROR An error occured while showing notification. An error occured while accessing https://api.github.com/repos/lightstep/community-team/issues/45: Not Found
1581719798
2020-02-14 19:12:54 ERROR Cannot save last shown commit date. 2020-02-14 19:12:54 ERROR An error occured while showing notification. An error occured while accessing https://api.github.com/repos/lightstep/community-team/issues/45: Not Found
1581719798
2020-02-14 19:13:56 ERROR Cannot save last shown commit date. 2020-02-14 19:13:56 ERROR An error occured while showing notification. An error occured while accessing https://api.github.com/repos/lightstep/community-team/issues/45: Not Found
1581719798
2020-02-14 19:14:57 ERROR Cannot save last shown commit date. 2020-02-14 19:14:57 ERROR An error occured while showing notification. An error occured while accessing https://api.github.com/repos/lightstep/community-team/issues/45: Not Found
1581719798
2020-02-14 19:15:58 ERROR Cannot save last shown commit date. 2020-02-14 19:15:58 ERROR An error occured while showing notification. An error occured while accessing https://api.github.com/repos/lightstep/community-team/issues/45: Not Found
1581719798
2020-02-14 19:17:00 ERROR Cannot save last shown commit date. 2020-02-14 19:17:00 ERROR An error occured while showing notification. An error occured while accessing https://api.github.com/repos/lightstep/community-team/issues/45: Not Found
1581719798
2020-02-14 19:18:01 ERROR Cannot save last shown commit date. 2020-02-14 19:18:01 ERROR An error occured while showing notification. An error occured while accessing https://api.github.com/repos/lightstep/community-team/issues/45: Not Found
1581719798
2020-02-14 19:19:02 ERROR Cannot save last shown commit date. 2020-02-14 19:19:02 ERROR An error occured while showing notification. An error occured while accessing https://api.github.com/repos/lightstep/community-team/issues/45: Not Found
1581719798
```

**Comment by sargsyan** [comment]
Could you please update the application with `brew upgrade github-notifier` and see if it works correctly with the new version?

**Comment by safris** [comment]
Hi @sargsyan, it seems after the last upgrade the `github-notifier` stopped notifying completely.

**Comment by sargsyan** [comment]
Hi @safris, my bad :( The correct commands for the update are:

```
brew upgrade github-notifier
github-notifier-install
```

So, now you need to call `github-notifier-install`

**Comment by safris** [comment]
Hi @sargsyan, I actually did that yesterday before I posted [this comment](https://github.com/sargsyan/github-notifier/issues/19#issuecomment-587244357). Today I woke up, and saw a notification from `github-notifier` re all missed GitHub notifications. I believe it is working now.

**Comment by sargsyan** [comment]
Great news! It will work with one caveat. I introduced a bug with the fix which can be fixed by another brew upgrade procedure.

**Comment by safris** [comment]
Hi @sargsyan, the notifier was working fine, until I started receiving the following:
<img width="332" alt="Screen Shot 2020-02-18 at 8 23 14 PM" src="https://user-images.githubusercontent.com/1258414/74801803-bee81000-528c-11ea-8e2a-b89dc1324610.png">
I have upgraded the notifier, and will see whether it fixes the issue.

**Comment by sargsyan** [comment]
Hi @safris, If you confirm at everything is good now, I will close the issue.

**Comment by safris** [comment]
Hi @sargsyan, so far so good! Thank you for this super useful tool, and for the support!

**Comment by safris** [comment]
Hi @sargsyan, today this started happening again:
<img width="333" alt="Screen Shot 2020-03-02 at 10 03 30 PM" src="https://user-images.githubusercontent.com/1258414/75747441-cae2c180-5cd1-11ea-84d6-0680ddb0d11f.png">

**Comment by sargsyan** [comment]
Hi @safris 

Could you update one more time? The last version 1.0.6 should have it completely fixed.

**Comment by safris** [comment]
Hi @sargsyan, I haven't seen this issue in about a week. There is, however, another thing that started happening. Often I get messages that say this:

<img width="337" alt="Screen Shot 2020-03-12 at 6 58 28 AM" src="https://user-images.githubusercontent.com/1258414/76474726-18c49b80-642f-11ea-98cd-092a1c3e2493.png">

Do you want me to open another ticket for this issue?

**Comment by sargsyan** [comment]
Hi @safris ,

I added the "Failed yo fetch a notification" with the fix of this issue. I thought that the errors should not happen frequently. I think they are happening because details about a private Github repository are not accessible with the token.  It seems that Github notification API tells that they are new actions in a private 'community-team' repository. However, when github-notifier follows the details link it fails because it has no access to private repositories.

The good news is that you can suspend the "Failed yo fetch a notification" notifications. To do that, you can open the file `/usr/local/Cellar/github-notifier/1.0.6/lib/github_notif` and modify the line 6
 
`readonly NOTIFY_ABOUT_FAILED_NOTIFICATIONS=0`

**Comment by safris** [comment]
Hi @sargsyan, thanks for the advice re disabling the notification. Is there fundamentally no way to get the details from a private repo via the token? If so, it definitely seems like a limitation in the Github API.

**Comment by sargsyan** [comment]
Hi @Safris, when I will have a time I will look at the problem with private repositories. I can post here if there will be updates on this. 

If the problem with the repeating notifications is solved we can close this issue.

**Comment by safris** [comment]
Hi @sargsyan, yes issue with repeating notifications is resolved. Thank you for this awesome utility!
